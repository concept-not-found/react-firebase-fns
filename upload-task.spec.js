const {createElement} = require('react')
const {create} = require('react-test-renderer')

const FirebaseMock = require('./firebase-mock')
const UploadTaskMock = require('./upload-task-mock')
const UploadTaskFactory = require('./upload-task')

describe('user', () => {
  it('should render {} initially', () => {
    const {Firebase} = FirebaseMock()
    const {UploadTask: uploadTask} = UploadTaskMock()
    const UploadTask = UploadTaskFactory(Firebase)
    const component = create(createElement(UploadTask, {uploadTask},
      (snapshot) => `snapshot: ${JSON.stringify(snapshot)}`
    ))
    expect(component.toJSON())
      .toMatchSnapshot()
  })

  it('when state changes, snapshot is rendered', () => {
    const {Firebase} = FirebaseMock()
    const {onMock, UploadTask: uploadTask} = UploadTaskMock()
    const UploadTask = UploadTaskFactory(Firebase)
    const component = create(createElement(UploadTask, {uploadTask},
      (snapshot) => `snapshot: ${JSON.stringify(snapshot)}`
    ))
    const listener = onMock.mock.calls[0][1]
    listener({
      bytesTransferred: 'some bytes transferred',
      totalBytes: 'some total bytes'
    })
    expect(component.toJSON())
      .toMatchSnapshot()
  })
})
