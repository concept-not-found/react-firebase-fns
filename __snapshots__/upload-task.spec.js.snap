// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user should render {} initially 1`] = `"snapshot: {}"`;

exports[`user when state changes, snapshot is rendered 1`] = `"snapshot: {\\"bytesTransferred\\":\\"some bytes transferred\\",\\"totalBytes\\":\\"some total bytes\\"}"`;
